### DISCUSSION TITLE
Linux desktop file and metadata

### DISCUSSION THREAD

**Discussion by rderooy** [86Box/86Box#2239] [Show and tell]
I created an RPM package on Fedora COPR for x86_64 and aarch64, including a desktop file and metadata file for the software center.
https://copr.fedorainfracloud.org/coprs/rob72/86Box/

It is not really suited for general use at this point, due to the fact that the roms cannot be found. As 86Box is currently hard-coded to find the `roms` directory in a subdirectory where the binary is located, this is not acceptable as that would end up being `/usr/bin/roms`.

Also, the fedora policies prohibit me creating a roms package and hosting it on copr, so that would have to be user installed, once it is possible to have roms in your home directory.

I can submit the desktop/metadata/spec files to the repo here, but where should I place them? Is something like `src/unix/assets` ok?

I also needed some suitable sized desktop icons. Right now my RPM spec file just converts the icon from the `src/win/assets` directory during build time to suitable sizes, but since that requires ImageMagick, I don't think it would be suitable to depend on it in tree, so I would probably also need to add some different sized PNG files.

NOTE: one annoying thing is that the metadata XML and RPM SPEC file contain version references, which will need to be updated for each release. For the XML you can either just always show the latest version, or you can make a list (most recent must be at the top). Not sure how to automate that with a cmake project.

Some other issues...
- Because aarch64 must be build with `-DDYNAREC=off`, I had to special case it in the spec file. I feel it would be better if it could auto disable Dynarec in the aarch64 case, instead of just failing the build.
- It would be great if 86Box could automatically locate a GM soundfont for fluidsynth. On most Linux distributions, `/usr/share/soundfonts/default.sf2` should do the trick, and is normally the FluidR3 soundfont. But of course Ubuntu has to do it's own thing and you should therefore also check for `/usr/share/sounds/sf2/FluidR3_GM.sf2`as an alternative location.

**Comment by rderooy** [comment]
I have added the assets to my fork here: https://github.com/rderooy/86Box

**Comment by Cacodemon345** [comment]
You need to turn on the new dynarec as well when building for AArch64.

**Reply by rderooy** [reply]
Really? because I tried the latest code from git on aarch64 and it failed with an error that dynarec had to be disabled.

**Reply by Cacodemon345** [reply]
Look more closely in the CMakeLists.txt file.

**Reply by rderooy** [reply]
ahh ok, I missed that you meant NEW_DYNAREC. I thought that was not yet ready for use?

**Reply by Cacodemon345** [reply]
Yes it is ready for use on ARM64.

**Comment by rderooy** [comment]
I have pushed some updates for the RPM SPEC file.
- It now creates a roms sub-package (this version of the spec file will not be run on copr)
- it creates a hacky symlink in `/usr/bin` to allow it to work with the current hardcoded roms location
- arm and aarch64 now has new dynarec enabled
